File: 3_single_inheritance.py
class Vehicle:
    def __init__(self):
        self.name = ""

    def drive(self):
        print(self.name + " is driving")

class Car(Vehicle):
    def __init__(self, name):
        super().__init__()
        self.name = name

obj = Car("Car")
obj.drive()

File: 3_single_inheritance.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

class Vehicle{
    public:
        string name;
        void drive() {
            cout << name << " is driving" << endl;
        }
};

class Car: public Vehicle{
    public:
        Car(string name) {
            this->name = name;
        }
};
int main(){
    Car obj("Car");
    obj.drive();
    
    return 0;
}